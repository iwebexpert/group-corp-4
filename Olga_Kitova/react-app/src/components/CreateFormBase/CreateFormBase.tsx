import React from 'react'
import { FormControl, TextField } from '@mui/material'

type CreateFormBaseType = {
  arrayField: CreateFormBaseFieldType[]
}
type CreateFormBaseFieldType = {
  key?: Multiple
  label: string
  name: string
  type?: string
  value: Multiple
  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void
  multiline?: boolean
  minRows?: Multiple
  disabled?: boolean
}

export default function CreateFormBase(props: CreateFormBaseType) {
  const { arrayField } = props
  
  return (
    <FormControl
      fullWidth
      component="form"
      sx={{
        '& .MuiTextField-root': { m: 2 },
        display: 'flex',
        flexDirection: 'column',
      }}
    >
      {arrayField.map((obj: CreateFormBaseFieldType) => (
        <TextField
          key={obj.name}
          label={obj.label}
          name={obj.name}
          type={obj.type || 'text'}
          value={obj.value}
          onChange={obj.onChange}
          multiline={obj.multiline || false}
          minRows={obj.minRows || ''}
          disabled={obj.disabled || false}
        />
      ))}
    </FormControl>
  )
}
