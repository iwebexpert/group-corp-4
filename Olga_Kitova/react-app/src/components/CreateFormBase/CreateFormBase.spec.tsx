import React from 'react'
import * as renderer from 'react-test-renderer'
import { test, expect } from '@jest/globals'
import CreateFormBase from './CreateFormBase'

const arrayFieldTest = [
  { key: '1', label: 'string_label', name: 'string_name', type: 'text', value: 'test' },
  { key: '2', label: 'string_label2', name: 'string_name2', type: 'text', value: 'test2' },
]
test('Testing with snapshots (CreateFormBase)', () => {
  const testRenderer = renderer.create(<CreateFormBase arrayField={arrayFieldTest} />)
  const treeJson = testRenderer.toJSON()
  expect(treeJson).toMatchSnapshot()
})
