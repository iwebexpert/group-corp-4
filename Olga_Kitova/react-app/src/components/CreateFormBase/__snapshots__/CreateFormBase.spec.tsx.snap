// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing with snapshots (CreateFormBase) 1`] = `
<form
  className="MuiFormControl-root MuiFormControl-fullWidth css-7wnvww-MuiFormControl-root"
>
  <div
    className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
  >
    <label
      className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={true}
    >
      string_label
    </label>
    <div
      className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        name="string_name"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value="test"
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-1in441m"
        >
          <span>
            string_label
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <div
    className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
  >
    <label
      className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={true}
    >
      string_label2
    </label>
    <div
      className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        name="string_name2"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value="test2"
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-1in441m"
        >
          <span>
            string_label2
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</form>
`;
